shared/actions/git/index.js

```diff
@@ -121,6 +122,31 @@ function* _setError(action: Constants.SetError): SagaGenerator<any, any> {
   yield put(Entities.replaceEntity(['git'], I.Map([['error', action.payload.gitError]])))
 }
 
+function* _badgeAppForGit(action: Constants.BadgeAppForGit): SagaGenerator<any, any> {
+  yield put(Entities.replaceEntity(['git'], I.Map([['isNew', I.Set(action.payload.ids)]])))
+}
+
+let _wasOnGitTab = false
+function* _onTabChange(action: RouteTreeConstants.SwitchTo): SagaGenerator<any, any> {
```

Comment:
Maybe it would be useful to know what path we came from so we wouldn't need this mutable state of `_wasOnGitTab`